pkg/sdk/testint/api_integrations_gen_integration_test.go

```diff
@@ -0,0 +1,436 @@
+	}
+
+	createAzureApiIntegration := func(t *testing.T) *sdk.ApiIntegration {
+		t.Helper()
+		return createApiIntegrationWithRequest(t, createApiIntegrationAzureRequest(t))
+	}
+
+	createGoogleApiIntegration := func(t *testing.T) *sdk.ApiIntegration {
+		t.Helper()
+		return createApiIntegrationWithRequest(t, createApiIntegrationGoogleRequest(t))
+	}
+
+	t.Run("create api integration: aws basic", func(t *testing.T) {
+		request := createApiIntegrationAwsRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: azure basic", func(t *testing.T) {
+		request := createApiIntegrationAzureRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: google basic", func(t *testing.T) {
+		request := createApiIntegrationGoogleRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: aws more options", func(t *testing.T) {
+		request := createApiIntegrationAwsRequest(t)
+
+		request = request.
+			WithAwsApiProviderParams(request.AwsApiProviderParams.WithApiKey(sdk.String("key"))).
+			WithApiBlockedPrefixes(prefixes(awsBlockedPrefix)).
+			WithComment(sdk.String("comment"))
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "comment")
+	})
+
+	t.Run("create api integration: azure more options", func(t *testing.T) {
+		request := createApiIntegrationAzureRequest(t)
+
+		request = request.
+			WithAzureApiProviderParams(request.AzureApiProviderParams.WithApiKey(sdk.String("key"))).
+			WithApiBlockedPrefixes(prefixes(azureBlockedPrefix)).
+			WithComment(sdk.String("comment"))
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "comment")
+	})
+
+	t.Run("create api integration: google more options", func(t *testing.T) {
+		request := createApiIntegrationGoogleRequest(t).
+			WithApiBlockedPrefixes(prefixes(googleBlockedPrefix)).
+			WithComment(sdk.String("comment"))
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "comment")
+	})
+
+	t.Run("alter api integration: aws", func(t *testing.T) {
+		integration := createAwsApiIntegration(t)
+
+		otherRoleArn := "arn:aws:iam::000000000001:/role/other"
+		setRequest := sdk.NewAlterApiIntegrationRequest(integration.ID()).
+			WithSet(
+				sdk.NewApiIntegrationSetRequest().
+					WithAwsParams(sdk.NewSetAwsApiParamsRequest().WithApiAwsRoleArn(sdk.String(otherRoleArn)).WithApiKey(sdk.String("key"))).
+					WithEnabled(sdk.Bool(true)).
+					WithApiAllowedPrefixes(prefixes(awsBlockedPrefix)).
```

Comment:
it doesn't matter, but you mixed allowed prefixes argument with blocked prefix value (in other tests too)

Reply:
It's on purpose because by default it's created with allowed, so to really alter something the blocked ones are used, maybe I can rename them to make it easier to read.